nuitka/freezer/DllDependenciesMacOS.py

```diff
@@ -252,38 +267,40 @@ def fixupBinaryDLLPathsMacOS(
     binary_filename, package_name, original_location, standalone_entry_points
 ):
     """For macOS, the binary needs to be told to use relative DLL paths"""
+    try:
+        had_self, rpath_map = detectBinaryPathDLLsMacOS(
+            original_dir=os.path.dirname(original_location),
+            binary_filename=original_location,
+            package_name=package_name,
+            keep_unresolved=True,
+            recursive=False,
+        )
+    except NuitkaForbiddenDLLEncounter:
+        inclusion_logger.info("Not copying forbidden DLL '%s'." % binary_filename)
+    else:
+        mapping = []
 
-    had_self, rpath_map = detectBinaryPathDLLsMacOS(
-        original_dir=os.path.dirname(original_location),
-        binary_filename=original_location,
-        package_name=package_name,
-        keep_unresolved=True,
-        recursive=False,
-    )
-
-    mapping = []
-
-    for resolved_filename, rpath_filename in rpath_map.items():
-        for standalone_entry_point in standalone_entry_points:
-            if resolved_filename == standalone_entry_point.source_path:
-                dist_path = standalone_entry_point.dest_path
-                break
-        else:
-            dist_path = None
-
-        if dist_path is None:
-            inclusion_logger.sysexit(
-                """\
-Error, problem with dependency scan of '%s' with '%s' please report the bug."""
-                % (original_location, rpath_filename)
+        for resolved_filename, rpath_filename in rpath_map.items():
```

Comment:
I think I am not getting what is going on here, why do we catch and swallow here, when we could just not raise. The DLL is also not forbidden in my mind, which is what that exception was for, but missing. 

I think what we should become able to, is tell outside code, this one wasn't found, but something told us, that's OK, so we remove it maybe in the `install_name_tool` call of things that use it.

Reply:
I got an error if I didn't use this.

```
Traceback (most recent call last):
  File "/Users/dryarullin/reps/Nuitka/nuitka/__main__.py", line 151, in <module>
    main()
  File "/Users/dryarullin/reps/Nuitka/nuitka/__main__.py", line 137, in main
    MainControl.main()
  File "/Users/dryarullin/reps/Nuitka/nuitka/MainControl.py", line 1090, in main
    _main()
  File "/Users/dryarullin/reps/Nuitka/nuitka/MainControl.py", line 1000, in _main
    copyDllsUsed(
  File "/Users/dryarullin/reps/Nuitka/nuitka/freezer/Standalone.py", line 659, in copyDllsUsed
    fixupBinaryDLLPathsMacOS(
  File "/Users/dryarullin/reps/Nuitka/nuitka/freezer/DllDependenciesMacOS.py", line 266, in fixupBinaryDLLPathsMacOS
    had_self, rpath_map = detectBinaryPathDLLsMacOS(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dryarullin/reps/Nuitka/nuitka/freezer/DllDependenciesMacOS.py", line 89, in detectBinaryPathDLLsMacOS
    had_self, resolved_result = _resolveBinaryPathDLLsMacOS(
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dryarullin/reps/Nuitka/nuitka/freezer/DllDependenciesMacOS.py", line 217, in _resolveBinaryPathDLLsMacOS
    raise NuitkaForbiddenDLLEncounter(binary_filename, plugin_name)
nuitka.Errors.NuitkaForbiddenDLLEncounter: ('/Users/dryarullin/miniconda3/envs/pyqt_new/lib/python3.11/site-packages/PySide6/Qt/qml/QtQuick/Effects/libeffectsplugin.dylib', 'pyside6')
```

Reply:
I think `fixupBinaryDLLPathsMacOS` ought to be reuse previous results, and not scan again. That is a weakness of the current code. For macOS there is this extra step to do the  "fixup". I need to look into making it so that it will not call `_resolveBinaryPathDLLsMacOS` twice. I believe that would be the proper place to cache.

Thanks for your hard work on this. This is hard stuff.